File: Ruby/day11.rb
arr = [
  [-1, -1, 0, -9, -2, -2],
  [-2, -1, -6, -8, -2, -5],
  [-1, -1, -1, -2, -3, -4],
  [-1, -9, -2, -4, -4, -5],
  [-7, -3, -3, -2, -9, -9],
  [-1, -3, -1, -2, -4, -5]
]

new_all = []
4.times do |r|
  4.times do |c|
    3.times do |s|
      3.times do |i|
        new_all << arr[s + r][i + c]
      end
    end
  end
end

new_list = new_all.each_slice(9).to_a

final_list = []
for item in new_list
  item.delete_at(3)
  item.delete_at(4)
  final_list << item.sum
end
puts final_list.max

File: Python/day11.py
a = [
    [-1, -1, 0, -9, -2, -2],
    [-2, -1, -6, -8, -2, -5],
    [-1, -1, -1, -2, -3, -4],
    [-1, -9, -2, -4, -4, -5],
    [-7, -3, -3, -2, -9, -9],
    [-1, -3, -1, -2, -4, -5],
]
n = []
for b in range(4):
    for k in range(4):
        for i in range(3):
            for j in range(3):
                n.append(a[i + b][j + k])


def split(arr, size):
    arrs = []
    while len(arr) > size:
        pice = arr[:size]
        arrs.append(pice)
        arr = arr[size:]
    arrs.append(arr)
    return arrs


final_array = split(n, 9)

new_list = []
for list in final_array:

    list.pop(3)
    list.pop(4)
    new_list.append(sum(list))

print(max(new_list))
